### DISCUSSION TITLE
Why do I get a "useForm is not a function" error when using React Hook forms in a Next JS project?

### DISCUSSION THREAD

**Discussion by s1egh4rdt** [vercel/next.js#59483] [Help] [answered]
### Summary

Hello, I encountered a problem when using React Hook Forms in my Next JS project.

I get an error: Error: (0 , react_hook_form__WEBPACK_IMPORTED_MODULE_1__.useForm) is not a function
![2023-12-11_13-35-26](https://github.com/vercel/next.js/assets/19279983/fdaaaa67-6d7d-4ad8-a9bf-e910c1958c90)

I can't figure out what the problem is. Why useForm doesn't work?
I have tried to uninstall the node_modules directory. None of these worked.

### Additional information

```js
Code form - LoginForm.tsx


import type {NextPage} from 'next'
import { useForm, SubmitHandler } from "react-hook-form";
import styles from './Auth.module.scss';

interface IFormInput {
  email: string
  password: string
}

export const LoginForm: NextPage = () => {
  const { register, handleSubmit } = useForm<IFormInput>();
  const onSubmit: SubmitHandler<IFormInput> = (data) => console.log(data);

  return (
    <div className={styles.formBlock}>
      <h1 className={styles.title}>Login</h1>
      <div className={styles.form_container}>
        <form onSubmit={handleSubmit(onSubmit)}>
            <label htmlFor="email">E-mail</label>
            <input type="email" placeholder="E-mail" 
              {...register('email', {
                required: "Email is required.",
                pattern: {
                  value: /^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}$/,
                  message: "Email must be correctly"
                }
              })}
            />
            <label htmlFor="password">Password</label>
            <input type="password"  
              {...register('password', {
                required: "Password is required..",
              })}
              />
            <input type="submit" />
        </form>
      </div>
    </div>
  )
}
```

**Comment by kerimhudson** [comment, marked as answer]
I was trying to figure this out too. Need to set `'use client'`

**Reply by ivozayas** [reply]
im so dumb

**Reply by lucasfelipedonascimento** [reply]
Funcionou aqui, valeu

**Reply by Yusuf-Najeeb** [reply]
You saved me a lot of time
Thanks

**Reply by mattheskaiser** [reply]
Thank you for saving my afternoon :)

**Reply by Kaddesh** [reply]
Mehn, this helped. Thank you

**Reply by Rell-Developer** [reply]
Thank you, sometimes forgot it XD

**Reply by erickwnilton** [reply]
thank you,  @kerimhudson

**Reply by drew-chidi** [reply]
Thank you. I almost cried 😀

**Reply by raunak-mishraa** [reply]
I spent too much figuring out this but couldn't, thank you so much man! @kerimhudson

**Reply by victornwakpa** [reply]
Thank you

**Reply by DzGunner** [reply]
Thank you :)

**Reply by manavbabber** [reply]
💯

**Reply by Sohad-Almadhoon** [reply]
Thank you soooooo much!

**Reply by sherozbek1706** [reply]
**Thanks bro, Also ChatGPT not anwered this question ?**

**Reply by Ola-Oluwajuwon** [reply]
Thank you!!! You saved me from a 3 days headache.
<img width="715" alt="errors" src="https://github.com/user-attachments/assets/21be9e26-65ea-48b4-aeb0-6f0ef253affc" />

**Reply by mattheskaiser** [reply]
Thank you!

**Reply by guilhermescr** [reply]
Thanks!!

**Comment by baramsivaramireddy** [comment]
thank you .

**Reply by shola-MTN** [reply]
yeeeee😫,

**Comment by ghost2023** [comment]
> I was trying to figure this out too. Need to set `'use client'`

I face palmed when reading this

**Comment by Luyolop1** [comment]
here i used "use-client" and still got problem.

'use-client'
import React from 'react';
import { PhoneIcon,EnvelopeIcon,MapPinIcon } from '@heroicons/react/24/solid';
import { useForm, SubmitHandler } from 'react-hook-form';

type Inputs = {
  name: string
  email: string;
  subject: string
  message: string
}

type Props = {}

const Contact = (props: Props) => {
  
  const {register,handleSubmit} = useForm<Inputs>()

  const onSubmit: SubmitHandler<Inputs> = (formData) => {
    window.location.href = `mailtoemaildummy@gmail.com?subject=${formData.subject}&body=Hi name is ${formData.name}. ${formData.message} 
    ${formData.email}`;
  }

  return (
    <div className='h-screen relative flex overflow-hidden flex-col text-left md:flex-row
    max-w-full justify-evenly mx-auto items-center z-0'>
      <h3 className='absolute top-24 uppercase tracking-[20px] text-gray-500 text 2xl'>
       Contact
       </h3>
      
      <div className='flex flex-col space-y-10'>
        <h4 className='text-4xl font-semibold text-center'>
            I have got just what you need.{" "}
            <span className='decoration-[#F7AB0A]/50 underline'>Let's talk</span>
        </h4>

        <div className='flex items-center space-x-5 justify-center'>
       <PhoneIcon className='text-[#f7AB0A] h-7 w-7 animate-pulse'/>
            <p className='text-2xl'>+27 60 499 2309</p>
        </div>

        <div className='flex items-center space-x-5 justify-center'>
            <EnvelopeIcon className='text-[#f7AB0A] h-7 w-7 animate-pulse'/>
            <p className='text-2xl'></p>
        </div>
        
        <div className='flex items-center space-x-5 justify-center'>
            <MapPinIcon className='text-[#f7AB0A] h-7 w-7 animate-pulse'/>
            <p className='text-2xl'>Cape Town. 8001</p>
        </div>

      <form onSubmit={handleSubmit(onSubmit)} className='flex flex-col space-y-2 w-fit mx-auto'>
        <div className='flex space-x-2'>
          <input {...register('name')} placeholder="Name" className="contactInput" type="text"/>
          <input {...register('email')}  placeholder="Email"className="contactInput"type="text"/>
        </div>

        <input  {...register('subject')} placeholder="subject"className="contactInput" type='text'/>

        <textarea {...register('message')}  className="contactInput" placeholder='message'/>
        <button className="bg-[#F7AB0A] py-5 px-10 rounded-md text-blank font-bold
        text-lg" type='submit'>Submit</button>
      </form>


      </div>
   </div>
  )
}

export default Contact

**Reply by bkostrowiecki** [reply]
@Luyolop1  You should type `use client`, not `use-client`. You don't need to put a hyphen in the middle.

**Reply by Walter28** [reply]
u saved my 2h of tryin to figure what had goes wrong thx bruhhh

**Comment by Luyolop1** [comment]
😅what a silly mistake idk where i got that from thank chief.🙏

**Comment by Luyolop1** [comment]
Thank you brother

**Comment by bryandandan** [comment]
For me, even with 'use client' at the top, it's not working :(
`const [form] = Form.useForm()` is breaking my next js build

**Reply by simsketch** [reply]
Same, pretty much boilerplate code too.

**Reply by Jore** [reply]
I'm having the same problem. Very frustrating.

**Reply by bkostrowiecki** [reply]
What library are you using here?

**Comment by mdmahfuzrp** [comment]
In NextJS 14, when using useForm from React Hook Form, the component must be marked as a client component. To ensure this, add "use client"; at the top of your file. This directive ensures that the component is rendered on the client side, which is necessary for React Hook Form to function correctly.

**Comment by pranta5** [comment]
Next.js 13+ (with the App Router) runs components server-side by default. Since react-hook-form depends on the browser (window object), it must run client-side. Adding "use client" forces the component to be treated as a client component.
